// randex/lib/test/markdown.test.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var custom_1 = require("../custom");
describe("markdown", function () {
    it("random", function () {
        var value = custom_1.Randex.random({
            set: "english",
            length: 3,
        });
        expect(value.length).toBe(3);
        value = custom_1.Randex.random(["english", 3]);
        expect(value.length).toBe(3);
        //
        value = custom_1.Randex.random({
            set: ["english", "lower"],
            length: 3,
        });
        expect(value.length).toBe(3);
        value = custom_1.Randex.random([["english", "lower"], 3]);
        expect(value.length).toBe(3);
        //
        value = custom_1.Randex.random({
            set: "english",
            length: [1, 3],
        });
        expect(value.length).toBeLessThanOrEqual(3);
        value = custom_1.Randex.random(["english", [1, 3]]);
        expect(value.length).toBeLessThanOrEqual(3);
        //
        value = custom_1.Randex.random({
            set: "number",
            length: 3,
        });
        expect(value.length).toBe(3);
        value = custom_1.Randex.random(["number", 3]);
        expect(value.length).toBe(3);
        //
        value = custom_1.Randex.random({
            charRange: "abc123",
            length: 3,
        });
        expect(value.length).toBe(3);
        //
        value = custom_1.Randex.random({
            set: ["english", "number"],
            length: 3,
        });
        expect(value.length).toBe(3);
        value = custom_1.Randex.random([["english", "number"], 3]);
        expect(value.length).toBe(3);
        //
        value = custom_1.Randex.random({
            set: "english",
            length: 2,
        }, {
            set: ["english", "number"],
            length: 2,
        });
        expect(value.length).toBe(4);
        value = custom_1.Randex.random(["english", 2], [["english", "number"], 2]);
        expect(value.length).toBe(4);
    });
    it("randomFileName", function () {
        var value = custom_1.Randex.fileName();
        value = custom_1.Randex.fileName({
            extension: "txt",
        });
        value = custom_1.Randex.fileName("txt");
        value = custom_1.Randex.fileName({
            fileNameLength: [7, 10],
            extensionLength: 5,
        });
        value = custom_1.Randex.fileName([7, 10], 5);
        value = custom_1.Randex.fileName({
            fileNameLength: 8,
            extension: "xml",
        });
        value = custom_1.Randex.fileName(8, "xml");
        expect(value).toBeTruthy();
    });
    it("randomUsername", function () {
        var value = custom_1.Randex.username();
        expect(value).toBeTruthy();
    });
    it("randomEmail", function () {
        var value = custom_1.Randex.email();
        value = custom_1.Randex.email({
            prefixLength: 8,
        });
        value = custom_1.Randex.email(8);
        value = custom_1.Randex.email({
            prefixLength: 8,
            lowDomainLength: 4,
            hightDomainLength: 2,
        });
        value = custom_1.Randex.email(8, 4, 2);
        value = custom_1.Randex.email({
            domain: "test.com",
        });
        value = custom_1.Randex.email("test.com");
        expect(value).toBeTruthy();
    });
    it("singleName", function () {
        var value = custom_1.Randex.singleName();
        expect(value).toBeTruthy();
        value = custom_1.Randex.singleName({ alphabet: "french" });
        expect(value).toBeTruthy();
        value = custom_1.Randex.singleName({ length: 5 });
        expect(value).toBeTruthy();
        value = custom_1.Randex.singleName({ length: [2, 5] });
        expect(value).toBeTruthy();
    });
    it("randomFullName", function () {
        var value = custom_1.Randex.fullName();
        expect(value).toBeTruthy();
    });
});
